src/utils/timestampUtils.tsx

```diff
@@ -0,0 +1,91 @@
+import TimeAgo from "javascript-time-ago";
+import en from "javascript-time-ago/locale/en";
+import { ReduxMessage } from "./../store/reducers/messageListReducer";
+import * as _ from "lodash";
+
+const userTimezone = Intl.DateTimeFormat().resolvedOptions().timeZone;
+
+TimeAgo.addLocale(en);
+const timeAgo = new TimeAgo(userTimezone);
+
+function formatMessageTime(
+  dateString: Date,
+  lastMessage: boolean,
+  use24hTimeFormat: boolean
+) {
+  const date = new Date(dateString);
+  const currentDate = new Date();
+
+  if (lastMessage && currentDate.toDateString() !== date.toDateString()) {
+    // If it's not the same day, show date.
+    return date.toDateString();
+  } else if (currentDate.getTime() - date.getTime() >= 3 * 60 * 60 * 1000) {
+    if (use24hTimeFormat) {
+      return date.toLocaleTimeString([], {
+        hour: "2-digit",
+        minute: "2-digit",
+        hour12: false,
```

Comment:
this can be condensed to a single call like
```
return date.toLocaleTimeString([], {
        hour: "2-digit",
        minute: "2-digit",
        hour12: !use24hTimeFormat,
}
```

No need for an if.